redisvl/extensions/router/semantic.py

```diff
@@ -596,3 +620,200 @@ def to_yaml(self, file_path: str, overwrite: bool = True) -> None:
         with open(fp, "w") as f:
             yaml_data = self.to_dict()
             yaml.dump(yaml_data, f, sort_keys=False)
+
+    # reference methods
+
+    @classmethod
+    def add_route_references(
+        cls,
+        route_name: str,
+        references: Union[str, List[str]],
+        vectorizer: BaseVectorizer = HFTextVectorizer(),
+        router_name: str = "",
+        redis_client: Optional[Redis] = None,
+        redis_url: str = "",
+    ) -> List[str]:
+        """Add a reference(s) to an existing route.
+
+        Args:
+            router_name (str): The name of the router.
+            references (Union[str, List[str]]): The reference or list of references to add.
+            vectorizer (BaseVectorizer): The vectorizer used to embed the reference.
+
+        Returns:
+            List[str]: The list of added references keys.
+        """
+
+        if isinstance(references, str):
+            references = [references]
+
+        if not cls._cls_index:
+            if (redis_client or redis_url) and vectorizer and router_name:
+                # connect to existing index:
+                index = SearchIndex.from_existing(
+                    router_name, redis_client=redis_client, redis_url=redis_url
+                )
+            else:
+                raise ValueError(
+                    "If class not instantiated, must provide redis_client or redis_url and vectorizer"
+                )
+        else:
+            index = cls._cls_index
+            if cls._vectorizer is None:
+                raise ValueError("Vectorizer not properly created")
+            vectorizer = cls._vectorizer
+            router_name = cls._name
+
+        route_references: List[Dict[str, Any]] = []
+        keys: List[str] = []
+
+        # embed route references as a single batch
+        reference_vectors = vectorizer.embed_many(references, as_buffer=True)
+
+        # set route references
+        for i, reference in enumerate(references):
```

Comment:
isn't this implementing the same or similar logic as `_add_routes(...)`?

Reply:
Yes. The gotcha here is the route reference one takes advantage of `self.` and this one is based on `cls.` I decided against "functionizing" since there wasn't a 3rd case. This is an example of where the code becomes cleaner if we drop the class method idea (like we had talked about at one point).